### ISSUE TITLE
Github Create repository button

### ISSUE THREAD

**Issue by NEO-SPECTRUMAN** [JustOff/github-wc-polyfill#62]
Looks like Create repository button is unavailable with javascripts turn on
on latest serpent/basilisk 52.9.0
![2022-04-28_023122](https://user-images.githubusercontent.com/88002487/165646870-ffedf097-4faa-463b-85b9-574fe6440213.png)

with javascript disabled it available

**Comment by SeaHOH** [comment]
Relevant: #54 c0f5ff34009e1288a8138b2be2d235fec13f6206

---

[github-wc-polyfill-1.2.19b1.xpi](https://github.com/JustOff/github-wc-polyfill/releases/download/1.2.18/github-wc-polyfill-1.2.19b1.xpi)

**Comment by NEO-SPECTRUMAN** [comment]
> github-wc-polyfill-1.2.19b1.xpi
![2022-04-28_115247](https://user-images.githubusercontent.com/88002487/165713077-10854dc5-f3d7-4609-ad6a-e01d36aa37ca.png)
im not tried to patch it

github-wc-polyfill-1.2.18.3.xpi installs well

**Comment by NEO-SPECTRUMAN** [comment]
i tryed more several times 
and the same result

**Comment by SeaHOH** [comment]
How do you install the addon? Tell me step by step.

**Comment by NEO-SPECTRUMAN** [comment]
with click on xpi https://github.com/JustOff/github-wc-polyfill/releases
with click on allow

with save as
drag&drop

also i tryed to delete/disable previous version
but the same result

**Comment by NEO-SPECTRUMAN** [comment]
also i tried to seetup on new/clear browser
basilisk52-g4.8.win32-git-20220423-f94c0da-uxp-059e35a46-xpmod

**Comment by SeaHOH** [comment]
Seems right steps, I have no idea of this strange compatibility issue now.

**Comment by AroKol78** [comment]
my info on adding an add-on
`https://github.com/Feodor2/Mypal/issues/32#issuecomment-955677595`
`https://github.com/Feodor2/Mypal/issues/32#issuecomment-955709187`

**Comment by SeaHOH** [comment]
Thx, I got it. https://github.com/JustOff/github-wc-polyfill/issues/12#issuecomment-752933505

@NEO-SPECTRUMAN So, how do you installed the version 1.2.18.3 ?

**Comment by Vangelis66** [comment]
**Serpent 52**'s `appVersion` value always stays put at `52.9.0`; despite it sharing the same GUID (i.e. `{ec8030f7-c20a-464f-9b0e-13a3a9e97384}`) with **official** Basilisk (and Mozilla Firefox, for that matter), the corresponding `<em:targetApplication>` block inside the extension's `install.rdf` file prohibits, as-is, installation on Serpent:
```xml
    </em:targetApplication>
    <em:targetApplication>
      <Description>
        <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
        <em:minVersion>52.9.2020.10.05</em:minVersion>
        <em:maxVersion>52.9.2022.*</em:maxVersion>
      </Description>
    </em:targetApplication>
```
In any case, _the extension doesn't officially support Serpent 52_, see #21 ...

If one has a manually downloaded/archived .XPI file on disk, to install on **St52** one has to first modify the `install.rdf` file accordingly, 
```xml
    <!-- Serpent 52 -->
    <em:targetApplication>
      <Description>
        <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
        <em:minVersion>52.9.0</em:minVersion>
        <em:maxVersion>52.9.*</em:maxVersion>
      </Description>
    </em:targetApplication>
```
repackage to XPI (though an archiver like **7-zip** will allow for modifications from within the original XPI file...) and then drop on `about:addons` tab...

Once installed, the extension will **auto-update from GitHub** whenever **a release is being made**, because in that case what takes precedence is the [update.xml](https://raw.githubusercontent.com/JustOff/github-wc-polyfill/master/update.xml) file, which dictates: 
```xml
<em:targetApplication>
              <RDF:Description>
                <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
                <em:minVersion>52.9</em:minVersion>
                <em:maxVersion>52.*</em:maxVersion>
                <em:updateLink>https://github.com/JustOff/github-wc-polyfill/releases/download/1.2.18/github-wc-polyfill-1.2.18.3.xpi</em:updateLink>
              </RDF:Description>
            </em:targetApplication>
```
which means a manually installed extension version can auto-update up to (and including) `v1.12.8.3`, which is the last version the `update.xml` file points to currently...

The same explanation applies to a Serpent 52 user left-clicking the 1.2.18.3 XPI link in the releases section, 

![release](https://user-images.githubusercontent.com/9669492/165748494-694c0367-bbb8-4233-a172-2f289d082682.JPG)

The installation on St52 will go through, because **it's governed b**y `update.xml`: 

![release2](https://user-images.githubusercontent.com/9669492/165748984-025678cf-4845-4573-a47e-9c560041d443.JPG)

Trying to install any other version of the extension (under the 1.2.18 release) will yield the not-compatible issue posted by OP...

TL;DR: To install any other version on **Serpent 52** except for **the very last release**, you'd have to first download the XPI to disk (obviously this includes `beta` pre-releases like `github-wc-polyfill-1.2.19b1.xpi`), modify/edit locally as above, install manually the modified XPI file... 
AFAICR, this is not a new issue, this is how **it always worked for Serpent 52 users** 😜  ...

Kindest regards 😄

**Comment by NEO-SPECTRUMAN** [comment]
> it always worked for Serpent 52 users 😜 ...

so i checked 
github-wc-polyfill-1.2.9.xpi
то
github-wc-polyfill-1.2.18.xpi

and no one was installed

but i someway has install 1.2.18.x on previous versions of serpent %)
without any install.rdf changes



наречие | someway
-- | --



</body>
</html>
